test/tstDetailsKokkosExtStdAlgorithms.cpp

```diff
@@ -0,0 +1,168 @@
+  double const val = 3.;
+
+  Kokkos::View<double *, DeviceType> v("v", n);
+  KokkosExt::iota(space, v, val);
+  std::vector<double> v_ref(n);
+  std::iota(v_ref.begin(), v_ref.end(), val);
+  auto v_host = Kokkos::create_mirror_view(v);
+  Kokkos::deep_copy(v_host, v);
+  BOOST_TEST(v_ref == v_host, tt::per_element());
+
+  Kokkos::View<int[3], DeviceType> w("w");
+  KokkosExt::iota(space, w);
+  std::vector<int> w_ref = {0, 1, 2};
+  auto w_host = Kokkos::create_mirror_view(w);
+  Kokkos::deep_copy(w_host, w);
+  BOOST_TEST(w_ref == w_host, tt::per_element());
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(exclusive_scan, DeviceType, ARBORX_DEVICE_TYPES)
+{
+  namespace KokkosExt = ArborX::Details::KokkosExt;
+  using ExecutionSpace = typename DeviceType::execution_space;
+
+  ExecutionSpace space{};
+
+  int const n = 10;
+  Kokkos::View<int *, DeviceType> x("x", n);
+  std::vector<int> x_ref(n, 1);
+  x_ref.back() = 0;
+  auto x_host = Kokkos::create_mirror_view(x);
+  for (int i = 0; i < n; ++i)
+    x_host(i) = x_ref[i];
+  Kokkos::deep_copy(x, x_host);
+
+  Kokkos::View<int *, DeviceType> y("y", n);
+  KokkosExt::exclusive_scan(space, x, y);
+
+  std::vector<int> y_ref(n);
+  std::iota(y_ref.begin(), y_ref.end(), 0);
+  auto y_host = Kokkos::create_mirror_view(y);
+  Kokkos::deep_copy(y_host, y);
+  Kokkos::deep_copy(x_host, x);
+  BOOST_TEST(y_host == y_ref, tt::per_element());
+  BOOST_TEST(x_host == x_ref, tt::per_element());
+  // in-place
+  KokkosExt::exclusive_scan(space, x, x);
+  Kokkos::deep_copy(x_host, x);
+  BOOST_TEST(x_host == y_ref, tt::per_element());
+  int const m = 11;
+  BOOST_TEST(n != m);
+  Kokkos::View<int *, DeviceType> z("z", m);
+  BOOST_CHECK_THROW(KokkosExt::exclusive_scan(space, x, z),
+                    ArborX::SearchException);
+  Kokkos::View<double[3], DeviceType> v("v");
+  auto v_host = Kokkos::create_mirror_view(v);
+  v_host(0) = 1.;
+  v_host(1) = 1.;
+  v_host(2) = 0.;
+  Kokkos::deep_copy(v, v_host);
+  KokkosExt::exclusive_scan(space, v, v);
+  Kokkos::deep_copy(v_host, v);
+  std::vector<double> v_ref = {0., 1., 2.};
+  BOOST_TEST(v_host == v_ref, tt::per_element());
+  Kokkos::View<double *, DeviceType> w("w", 4);
+  BOOST_CHECK_THROW(KokkosExt::exclusive_scan(space, v, w),
+                    ArborX::SearchException);
+  v_host(0) = 1.;
+  v_host(1) = 0.;
+  v_host(2) = 0.;
+  Kokkos::deep_copy(v, v_host);
+  Kokkos::resize(w, 3);
+  KokkosExt::exclusive_scan(space, v, w);
+  auto w_host = Kokkos::create_mirror_view(w);
+  Kokkos::deep_copy(w_host, w);
+  std::vector<double> w_ref = {0., 1., 1.};
+  BOOST_TEST(w_host == w_ref, tt::per_element());
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(reduce, DeviceType, ARBORX_DEVICE_TYPES)
+{
+  using ExecutionSpace = typename DeviceType::execution_space;
+  ExecutionSpace space{};
+
+  namespace KokkosExt = ArborX::Details::KokkosExt;
+
+  Kokkos::View<int[6], DeviceType> v("v");
+  Kokkos::deep_copy(v, 5);
+  BOOST_TEST(KokkosExt::reduce(space, v, 3) == 33);
```

Comment:
add a using-declaration and do unqualified call here and below
```C++
  using ArborX::Details::KokkosExt::reduce;

Reply:
I prefer the way I did it.